src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -173,6 +180,26 @@ SimpleNameSyntax syntax
 			}
 		}
 
+		private static void AnalyzeConditionalImmutability( SyntaxNodeAnalysisContext ctx ) {

```

Comment:
Last suggestion: maybe rename to mention this is about methods?

`AnalyzeMethodDeclarationConditionalImmutabilityAnnotations` or something...